File: ruby/snakesladders/lib/snakesladders/models/snake.rb
module Snakesladders
  module Models
    class Snake
      attr_accessor :start_point, :end_point

      def initialize(start_point, end_point)
        self.start_point = start_point
        self.end_point = end_point

        validate_snake
      end

      private def validate_snake
        raise ::Snakesladders::Errors::InvalidLadderError.new("A snake should end at a lower position than start position") if end_point > start_point
      end
    end
  end
end

File: java/snakes-and-ladders/src/main/java/gagan93jtg/lld/models/Snake.java
package gagan93jtg.lld.models;

public class Snake {
  private int start;
  private int end;

  public Snake(int start, int end) {
    this.start = start;
    this.end = end;

    validateSnake();
  }

  public int getStart() {
    return start;
  }

  public int getEnd() {
    return end;
  }

  private void validateSnake() {
    if (end > start) {
      throw new IllegalArgumentException("A snake should end at a lower position than start position");
    }
  }
}
